src/main/java/com/aerospike/mapper/tools/virtuallist/ReactiveVirtualList.java

```diff
@@ -84,7 +85,13 @@ public ReactiveVirtualList(@NotNull IReactiveAeroMapper reactiveAeroMapper, @Not
 
         return reactiveAeroMapper.getReactorClient()
                 .operate(writePolicy, key, interactor.getOperation())
-                .map(keyRecord -> (E)interactor.getResult(keyRecord.record.getList(binName)));
+                .map(keyRecord -> {
+                    E result = keyRecord == null ? null : (E) interactor.getResult(keyRecord.record.getList(binName));
+                    if (result != null) {
+                        reactiveAeroMapper.getMappingConverter().resolveDependencies(ClassCache.getInstance().loadClass(result.getClass(), reactiveAeroMapper));
+                    }
+                    return result;
+                });
```

Comment:
Worth extracting this to a method, to avoid code duplication.

Reply:
Agree, done.